docs/LRM-feedback.md

```diff
@@ -0,0 +1,420 @@
+
+#### 4.13 Filter 
+
+The filter keyword takes any function that returns a boolean and applies it to elements of an array. This allows filter to quickly generate a new array that consists of elements that match whatever member criteria your function tests for.
+
+Example: `filter(f,arr1);`
+
+
+In the above example, arr1 contains an array of strings that are either `dog` or `cat`. func f returns true if the element is equal to `dog`. The above expression would return an array only consisting of every element in arr1 that contains `dog`.
+
+
+## 5: Declarations  
+
+Objects are instantiated via declarations, which explicitly assign a data type to a variable. In Fire a variable cannot be declared without also being assigned to a value. Types are explicit in FIRE. The format of a declaration is as follows:
+
+`{type} {variable name} = {value};`.
+
+#### 5.1 Data Types
+
+#### 5.1.1. `int`
+
+Identifiers of type `int` represent the natural numbers. The upper and lower bounds for int are defined by the architechtural constraints of the computer.
+
+Example:
+
+`int num = 32;`
+
+`int` types can also be assigned to the result of expressions:
+
+`int a = 34 * 2 + (2 / 1);`
+
+#### 5.1.2. `str`
+
+Identifiers of type `str` are used to represent sequences of characters, strings. Strings are immutable and can be declared in the following manner:
+
+Example:
+
+`str myString = "Hello World";`
+
+#### 5.1.3. `file`
+
+Files are regarded as first-class citizens in FIRE. This is made apparent by the importance and centrality of files. A `file` type represents either an existing file or a file that is to be written. This allows the programmer to more easily perform operations on the file.
+
+The syntax for instantiating a `file` object is as follows:
+
+`file f = file("filename.csv", "<mode>");`
+
+In the example provided above, two arguments are fed into the `file(...)` argument: *filename* for reading, writing or both, and *mode*. The *mode* argument can be `r` for read only, `w` for write only, and `rw` for both.
+
+Example:
+
+`file f = file("test.csv", "rw");` will open the File named test.csv in the current directory for both reading and writing.
+  
+`file f = file("filename.csv", "<mode>", "<delim>");`
+
+An optional third argument *delim* may be provided to the constructor specifying a delimiter for reading. If the *delim* argument is not supplied it will default to `\n`.
+
+Example:
+
+`file f = file("Program.java", "rw", ";");` will open the File named `Program.Java` in the current directory for both reading and writing. Calls to `read()` will read in chunks of the file delimited by the `;` character.
+
+#### 5.1.4. `func`
+
+`func` objects reference functions and are treated as first class citizens. The structure of `func` variable declarations is as follows:
+
+`func <return type> <name> = (<parameters>) => { <function body> };`
```

Comment:
can the function body inherit variables from the enclosing scope? that is, does it support closures?